src/ray/util/logging.cc

```diff
@@ -73,6 +81,62 @@ struct StdoutLogger : public google::base::Logger {
 static StdoutLogger stdout_logger_singleton;
 #endif
 
+#ifdef RAY_USE_SPDLOG
+/// NOTE(lingxuan.zlx): we reuse glog const_basename function from its utils.
+inline const char *ConstBasename(const char *filepath) {
+  const char *base = strrchr(filepath, '/');
+#ifdef OS_WINDOWS  // Look for either path separator in Windows
+  if (!base) base = strrchr(filepath, '\\');
+#endif
+  return base ? (base + 1) : filepath;
+}
+
+class SpdLogMessage final {
+ public:
+  explicit SpdLogMessage(const char *file, int line, int loglevel) : loglevel_(loglevel) {
+    stream() << ConstBasename(file) << ":" << line << ": ";
+  }
+  inline std::shared_ptr<spdlog::logger> get_logger() {
+    auto logger = spdlog::get("ray_log_sink");
+    if (!logger) {
+      logger = spdlog::get("stderr");
+    }
+    // If no default logger we just emit all log informations to stderr.
+    if (!logger) {
+      logger = spdlog::stderr_color_mt("stderr");
+      logger->set_pattern("[%Y-%m-%d %H:%M:%S,%e %L %P %t] %v");
+    }
+    return logger;
+  }
+
+  inline void Flush() {
+    auto logger = get_logger();
+    // To avoid dump duplicated stacktrace with installed failure signal
+    // handler, we have to check whether glog failure signal handler is enabled.
+    if (!RayLog::IsFailureSignalHandlerEnabled() &&
+        loglevel_ == static_cast<int>(spdlog::level::critical)) {
+      stream() << "\n*** StackTrace Information ***\n" << ray::GetCallTrace();
+    }
+    logger->log(static_cast<spdlog::level::level_enum>(loglevel_), "{}", str_.str());
```

Comment:
```suggestion
    logger->log(static_cast<spdlog::level::level_enum>(loglevel_), /*fmt*/ "{}", str_.str());
```?

Reply:
Not sure if it is the correct argument. Can you specify what this argument means?